### DISCUSSION TITLE
RPGLE language tools 0.14.0 - huge release

### DISCUSSION THREAD

**Discussion by worksofliam** [codefori/vscode-ibmi#915] [Announcements]
This is the largest release for vscode-rpgle we've had in a while. Functionality hasn't changed, but the entire codebase got a re-write.

The extension is no longer using the VS Code API to provide language tools to the editor. It has been re-written to use the LSP. This means the language processing happens in [a separate process](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide). That is a large performance increase for RPGLE developers.

This is something that I should have done from the start. When I started developing this extension, I didn't expect it to turn into what it did, but I am glad to see where it is going.

### Noticeable changes

* The linter is enabled by default but won't be enabled until it finds the lint configuration file (`rpglint.json` member or local file). No more needing to turn it on via config.
* `Fix all auto-fixable problems` has been replaced with `Format Document`. This is the way other languages/LSP implementations do it.
* RPGLE fixed format column assist decorations still exist, but the fixed format prompt is gone. It can come back, but only if people complain about it missing.

### Going forward

Everything else should be the same. While I do plan on continuing to support those who are developing in source members, a large focus is encouraging users into using git - which usually means writing code on the local workspace. Project mode, coming in the future, will only work with local workspaces, not source members or streamfiles.

This is a large project and it was a large release. Of course there will be bugs! I can't wait for people to report them. I am determined to fix them quickly.

**Comment by rishabkaushik** [comment]
Hi @worksofliam, 

I am here to complain about the column-assist functionality especially the fixed format prompt that is gone.  I know, my use case can not be the only reason to bring the feature back. But since the community is grown substantially after this change was made, may be it's a good time to ask other or reflect back? It was helpful especially for the beginners(to IBM i world) maintaining old RPG code through VS Code.

Thanks

**Reply by miguelmestre** [reply]
My company is also willing to adopt VSCode however we have a lot of fixed format code that we still need to maintain.It would be great to have the prompt back...
I oppened an issue https://github.com/codefori/vscode-ibmi/issues/2145

Thanks
